lib/puppet/application/master.rb

```diff
@@ -172,7 +172,7 @@ def compile
         raise "Could not compile catalog for #{options[:node]}"
       end
 
-      puts PSON::pretty_generate(catalog.to_resource, :allow_nan => true, :max_nesting => false)
+      puts JSON::pretty_generate(catalog.to_resource, :allow_nan => true, :max_nesting => false)
```

Comment:
For future us, JSON calls [`to_json`](https://github.com/flori/json/blob/v1.8.3/ext/json/ext/generator/generator.c#L867-L870) on objects that are not JSON values (string, numeric, etc).